bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/StructuredSelection.java

```diff
@@ -215,6 +225,7 @@ public List toList() {
 	 */
 	@Override
 	public String toString() {
-		return isEmpty() ? JFaceResources.getString("<empty_selection>") : toList().toString(); //$NON-NLS-1$
+		return isEmpty() ? JFaceResources.getString("<empty_selection>") //$NON-NLS-1$
+				: stream().map(String::valueOf).collect(Collectors.joining(", ", "[", "]")); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
```

Comment:
I'd do `Arrays.toString(elements)` instead to simplify the code.

Reply:
@laeubi can you please address this suggestion?